File: example-blocking/src/main/java/com/davin111/exampleblocking/ContentController.java
package com.davin111.exampleblocking;

import com.davin111.exampleblocking.dto.ContentsDto;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class ContentController {

    private final ContentService contentService;

    @GetMapping("/v1/contents/free")
    public ContentsDto getContentsFree() {
        return contentService.getContentsFree();
    }

    @GetMapping("/v1/contents/premium")
    public ContentsDto getContentsPremium() {
        Long userId = 0L;
        return contentService.getContentsPremium(userId);
    }
}

File: example-nonblocking-kotlin/src/main/kotlin/com/davin111/examplenonblockingkotlin/ContentController.kt
package com.davin111.examplenonblockingkotlin

import com.davin111.examplenonblockingkotlin.dto.ContentsDto
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class ContentController(
    private val contentService: ContentService,
) {

    @GetMapping("/v1/contents/free")
    suspend fun contentsFree(): ContentsDto {
        return contentService.getContentsFree()
    }

    @GetMapping("/v1/contents/premium")
    suspend fun contentsPremium(): ContentsDto {
        val userId = 0L
        return contentService.getContentsPremium(userId)
    }
}
